include/modem/location.h

```diff
@@ -253,6 +253,11 @@ struct location_config {
 	 * the valid range is 10...65535 seconds.
 	 */
 	uint16_t interval;
+	/**
+	 * @brief Timeout (in milliseconds) of how long the entire location request can take.
+	 * SYS_FOREVER_MS means that the timer is disabled.
+	 */
+	int32_t timeout;
```

Comment:
you could mention the default value maybe

Reply:
I didn't mention because it's not anywhere else. We've discussed this with @tokangas few times and I'll try to find a solution at some point but not within this PR.
The main problem is that it's not enough to just say that default value is 5min because that's not set automatically anywhere but you need to call specific functions for getting the defaults applied. But we'll fix this in a way or another later this year.